### PR TITLE
Disable "wpcom-user-bootstrap" until it works correctly with GeoIP.

### PR DESCRIPTION
**Pull request by DanReyLop**
For more backstory on this change, see https://github.com/Automattic/wp-calypso/pull/16910 and p4TIVU-7BK-p2.

**TL;DR:** To make the initial loading of Calypso faster, a server-side request is made to the `/me` endpoint. Since that request comes from the Calypso server itself (geolocated in US), that means that every user will be effectively geolocated to US. There aren't many places when the user's country is checked, but the 2 I could fine were:
* GApps check. Since Google is banned in some countries, we need to check the user's country before allowing them to purchase GApps. That's effectively useless because the user will be geolocated in the US no matter what.
* Store on wp.com. We are rolling out the `Store (BETA)` feature to US and Canada business users only. Since everyone is geolocated in the US, we've received complaints of European people that saw the section and then couldn't use it because we're only supporting US/CA for now.

After dropping a bomb on the Calypso P2 and immediately going AFK for 3 weeks, I came back to the shocking revelation that the problem hadn't fixed itself :)

There are several options too solve this (I brainstormed a bit in the P2 post) but the immediate action is to disable `wpcom-user-bootstrap` so the problem is fixed. In hindsight, I should have made this change as soon as we detected the problem.

### CODE REVIEW

**Comment by yoavf** [comment]
Pinging @josephscott for thoughts about performance implications - I think /me is much faster that what it was when this was first setup, but would be nice to confirm that.

In any case we should probably still do this for now, but figure out a better solution.

**Comment by josephscott** [comment]
I didn't see any measurements listed, or methods for measuring the difference, so it is hard to say what sort of impact this will have.

Are all Calypso servers going to always be located in the US?

**Comment by yoavf** [comment]
@DanReyLop I didn't mean to block this by mentioning performance - but maye we can start by doing this on wpcalypso, and thenrun some concrete tests on performance to see what we're losing here, if at all. I'd look for time to first byte (which should theoretically improve, I think), but then also first and full render time (I'm not sure what the best metrics are for that)

**Comment by Tug** [comment]
Just a quick note that we should wait for https://github.com/Automattic/i18n-calypso/pull/40 to land before merging this as it may otherwise break internationalisation in calypso.

**Comment by Tug** [comment]
Btw, did we investigate setting `X-Forwarded-For` on the `/me` request instead?

**Comment by DanReyLop** [comment]
> Just a quick note that we should wait for Automattic/i18n-calypso#40 to land before merging this as it may otherwise break internationalisation in calypso.

👍 

> Btw, did we investigate setting X-Forwarded-For on the /me request instead?

We can't set the `X-Forwarded-For` header in the Calypso server, it will just be ignored. For more context, see the back-and-forth in pMz3w-7o0-p2

**Comment by DanReyLop** [comment]
@yoavf I've taken some measurements.

Setup: Local Docker build (`npm run build-docker && npm run docker`), pointing to the real `wordpress.com` (no sandbox). I used Chrome's `Fast 3G` network throttling preset. It has an even slower preset (`Slow 3G`), but in that one wp.com takes minutes to load.

With `wpcom-user-bootstrap` (`master`):
- Time to first byte: ~600ms
- Time to the "W" background (some styles loaded): ~23s
- DOMContentLoaded: ~30s
- onload: ~32s (at that point, the user's avatar is rendered)

Without `wpcom-user-bootstrap` (this branch):
- Time to first byte: ~570ms
- Time to the "W" background (some styles loaded): ~23s
- DOMContentLoaded: ~30s
- onload: ~32s
- Immediately after `onload`, the `/me` request is sent, which takes ~575ms. (3G has a lot of latency)

So, overall, I don't think the impact is significant. We're trading a few milliseconds in the first page load vs an extra HTTP request. Note that the extra `/me` request is sent in parallel with many others (`/me/preferences`, `/me/sites`, `/me/settings` twice for some reason, etc).

**Comment by DanReyLop** [comment]
Can I please get a final sign-off on this? Is the performance degradation acceptable?

**Comment by yoavf** [comment]
@DanReyLop sorry for the delay, was afk.  Can we just do this on wp-calypso and/or stage first, to confirm there are no issues related to the missing bootstrapped user? It'll be easier to test.

FWIW I'm mostly worried about the locale switch happening too late if the user doesn't have the locales files loaded already.
 
👍 for me to merge without the production change

**Comment by DanReyLop** [comment]
I've released it in `staging`, `wpcalypso` and `horizon`. What kind of test do you want to make? I'm not sure I follow what you said about the locale switch.

**Comment by yoavf** [comment]
@DanReyLop I'll test this tomorrow, but what I'm seeing right now is that the title of the page is untranslated in staging (and wp-calypso) vs translated in production.

If it's just the title - that's not a huge issue, but I think I'm generally seeing translations being loaded later - i.e the site loads in English and only a little while later (after /me completes) loads in the UI locale (if you have one).

This already happens in production today on the first load if you don't have the language files cached, but it seems that with this patch it may be happening all the time. But I'm not sure - this needs testing. If this turns out to be true, it may be enough to store the user locale in localStorage or in another way.

**Comment by yoavf** [comment]
>  But I'm not sure - this needs testing

Confirmed that with bootstrap off, the language switch happens too late. I think we should be able to work around that by saving the localeSlug of the user somewhere, but haven't found a solution yet.
